### PR TITLE
qutebrowser: add options and keybindings

### PR DESCRIPTION
**Pull request by iv-nn**
### Description

Following #1132 and #1149.

I added 3 settings directly as options since they require to be formatted as python dictionary: `aliases`, `url.searchengines` and `bindings.key_mappings`.
I didn't add `content.headers.custom`, `content.javascript.log` and `hints.selectors`, while they are the 3 other dictionaries they are also probably less used.

I'm not sure how to handle default value, I didn't put anything but qutebrowser do have some default values for `aliases` and `url.searchengines`. Should I redefine them to make them appear in the documentation? We would need to keep them in sync then, it would maybe even require to set them to empty dictionaries before.

I also added a keybindings option.

### Checklist

- [x] Code formatted with `./format`.

- [x] Code tested through `nix-shell --pure tests -A run.all`.

- [x] Test cases updated/added. See [example](https://github.com/rycee/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [x] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://github.com/rycee/home-manager/blob/master/CONTRIBUTING.md#commits) for more information and [recent commit messages](https://github.com/rycee/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [ ] Added myself as module maintainer. See [example](https://github.com/rycee/home-manager/blob/068ff76a10e95820f886ac46957edcff4e44621d/modules/programs/lesspipe.nix#L6).

  - [ ] Added myself and the module files to `.github/CODEOWNERS`.

### CODE REVIEW

File: modules/programs/qutebrowser.nix [hunk 1]
```diff
@@ -50,6 +82,90 @@ in {
       '';
     };
 
+    keyMappings = mkOption {
+      type = types.attrsOf types.str;
+      default = { };
+      description = ''
+        This setting can be used to map keys to other keys. When the key used as dictionary-key is pressed, the binding for the key used as dictionary-value is invoked instead. This is useful for global remappings of keys, for example to map Ctrl-[ to Escape. Note that when a key is bound (via <literal>bindings.default</literal> or <literal>bindings.commands</literal>), the mapping is ignored.
```
Comment (rycee) [inline modules/programs/qutebrowser.nix:89]: Please limit the description lines to 80 characters or so for consistency with other modules. The same applies for the other options.

File: modules/programs/qutebrowser.nix [hunk 2]
```diff
@@ -50,6 +82,90 @@ in {
...
+      '';
+    };
+
+    disableDefaultBindings = mkOption {
```
Comment (rycee) [inline modules/programs/qutebrowser.nix:93]: Better have `enableDefaultBindings` and default it to `true`.

**Comment by iv-nn** [comment]
What do you think about the default values, like [`url.searchengines`](https://qutebrowser.org/doc/help/settings.html#url.searchengines) default to 
```
{
    "DEFAULT" = "https://duckduckgo.com/?q={}"
}
```
Should I replicate this in the nix config ? This would add the benefit of documenting this default value in the home-manager doc but it could change at some point.

**Comment by rycee** [comment]
Thanks! Rebased to master in 03b4f81679456dc565722b38b18c27911b135d66.

**Comment by rycee** [comment]
Generally it's not a good idea to guess the defaults of a package since it depends on the version of the package. Better then to let the package itself choose what the defaults are.

Note in the merged code I changed `keybindings` to `keyBindings` to be consistent with `keyMappings`.

**Pull request closed without merging**